arlo-client/src/components/Audit/index.tsx

```diff
@@ -79,17 +74,15 @@ const Audit: React.FC<{}> = () => {
     'Participants'
   )
 
-  const menuItems = useMemo(
-    () =>
-      setupStages.map(
-        (s: ElementType<typeof setupStages>): ISidebarMenuItem => ({
-          title: s,
-          active: s === stage,
-          action: () => setStage(s),
-        })
-      ),
-    [stage]
-  )
+  const [menuItems, refresh] = useSetupMenuItems(stage, setStage)
+
+  useEffect(() => {
+    refresh()
+  }, [])
+
+  const activeStage = menuItems.find(m => m.title === stage)
+  const nextStage = menuItems[menuItems.indexOf(activeStage!) + 1]
+  const prevStage = menuItems[menuItems.indexOf(activeStage!) - 1]
```

Comment:
Any edge cases to consider here when `indexOf` is first or last element?

Reply:
> Any edge cases to consider here when `indexOf` is first or last element?

@benadida Right now, those functions are only accessible when they wouldn't be undefined, but that's not checked programmatically. I'll take a look and see if there's a place where it would be efficacious to check for that to prevent human error.

Reply:
@MorganLove are we talking about the same thing? I'm wondering what happens if `menuItems.indexOf(activeStage!)` returns `0`, and then `prevStage` is set to `menuItems[-1]`. Is that simply a null and it's ok?

Reply:
@benadida I believe we are. If it returns `0` then `prevStage` is `undefined`, but if the index of the active stage is 0, then that's the first item in the menu and the first item never gets passed `prevStage`, since it's meaningless. Therefore, nothing ever tries to access values on the object in a situation in which it would be undefined.

I just pushed a commit that makes it a bit more explicit, by marking the types in these lines as `| undefined` and then in the `Setup` component marking where we know it isn't undefined and commenting where we shouldn't be sending in the undefined values. Hope that helps. :slightly_smiling_face: